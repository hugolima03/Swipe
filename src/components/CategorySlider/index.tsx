import CategorySliderItem from 'components/CategorySliderItem'
import Slider, { SliderSettings } from 'components/Slider'

import * as S from './styles'

const settings: SliderSettings = {
  dots: false,
  arrows: false,
  vertical: false,
  verticalSwiping: false,
  infinite: true,
  centerMode: true,
  centerPadding: '60px'
}

const CategorySlider = () => (
  <S.Wrapper>
    <Slider settings={settings}>
      <CategorySliderItem />
      <CategorySliderItem />
      <CategorySliderItem />
    </Slider>
  </S.Wrapper>
)

export default CategorySlider
