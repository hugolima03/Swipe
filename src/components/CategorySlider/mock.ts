import { Asset } from 'generated/graphql'
import { CategorySliderProps } from '.'

export const mock = {
  categories: [
    {
      name: 'Math',
      id: '123',
      image: {
        __typename: 'Asset',
        url: 'https://source.unsplash.com/user/willianjusten/1042x580'
      } as Asset
    },
    {
      name: 'History',
      id: '123',
      image: {
        __typename: 'Asset',
        url: 'https://source.unsplash.com/user/willianjusten/1042x580'
      } as Asset
    }
  ]
} as CategorySliderProps
