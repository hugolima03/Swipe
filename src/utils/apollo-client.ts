import { ApolloClient, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: `${process.env.NEXT_PUBLIC_API_URL}`,
  cache: new InMemoryCache(),
  headers: { Authorization: `Bearer ${process.env.NEXT_PUBLIC_GRAPHCMS_TOKEN}` }
})

export default client
