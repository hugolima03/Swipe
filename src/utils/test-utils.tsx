import { MockedProvider, MockedResponse } from '@apollo/client/testing'

import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const mocks: MockedResponse<Record<string, unknown>>[] = []

const customRender = (
  ui: React.ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </MockedProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
