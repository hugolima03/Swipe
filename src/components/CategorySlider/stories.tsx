import { Story, Meta } from '@storybook/react'
import CategorySlider from '.'

export default {
  title: 'CategorySlider',
  component: CategorySlider,
  parameters: { layout: 'fullscreen' }
} as Meta

export const Default: Story = () => <CategorySlider />
