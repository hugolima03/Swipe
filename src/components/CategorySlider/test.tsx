import { render, screen } from 'utils/test-utils'

import CategorySlider from '.'

describe('<CategorySlider />', () => {
  it('should render the heading', () => {
    render(<CategorySlider />)
  })
})
