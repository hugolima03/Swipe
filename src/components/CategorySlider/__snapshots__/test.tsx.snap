// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategorySlider /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    CategorySlider
  </h1>
</main>
`;
